README.md

```diff
@@ -713,6 +713,24 @@ timestamps.
 
 [TCurl]: https://github.com/uber/tcurl
 
+## Non filesystem and asynchronous source loading
+
+You can pass an object with a custom `load` function as the `fs` argument to `new Thrift()`. The function provided will be passed file names and must call the callback function with the source associated with the file name. You can pass a callback function as second argument to `new Thrift()`, it will be called when the model is fully loaded or an error occured. This way, you can fetch thrift sources from a server for example. Here is an example :
+```javascript
+var request = require('request')
+var Thrift = require('thriftrw').Thrift
+
+function fetchSource(filename, cb) {
+  request('https://mysite.com/thrift/' + filename, function (err, res, body) {
+    cb(err, body)
+  });
+}
+
+new Thrift({fs: {load: fetchSource}}, function (err, thrift) {
+  console.log(err, thrift)
+})
+```
```

Comment:
Can we consider an alternative constructor for the asynchronous case?

```
Thrift.load(source, loader, function(err, rootThrift) {
});
```

Reply:
We know have a `Thrift.load` constructor